rust-runtime/aws-smithy-checksums/src/http.rs

```diff
@@ -47,16 +47,32 @@ pub trait HttpChecksum: Checksum + Send + Sync {
         header_map
     }
 
+    /// Either return this checksum as a http-1x `HeaderMap` containing one HTTP header, or return an error
+    /// describing why checksum calculation failed.
+    fn headers_http_1x(self: Box<Self>) -> http_1x::HeaderMap<http_1x::HeaderValue> {
+        let mut header_map = http_1x::HeaderMap::new();
+        header_map.insert(self.header_name(), self.header_value_http_1x());
+
+        header_map
+    }
+
     /// Return the `HeaderName` used to represent this checksum algorithm
     fn header_name(&self) -> &'static str;
 
-    /// Return the calculated checksum as a base64-encoded `HeaderValue`
+    /// Return the calculated checksum as a base64-encoded http-02x `HeaderValue`
```

Comment:
Given the internal facing nature of this crate (and the 0.x versioning) can we remove the legacy http all together or no?